### PR TITLE
Async Crypto Endeavour

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by diasdavid** [comment]
LGTM

File: test/browser.js [hunk 1]
```diff
@@ -2,9 +2,8 @@
 
 const w = require('webrtcsupport')
 
-require('./websockets-only')
+// require('./websockets-only')
```
Comment (diasdavid) [inline test/browser.js]: Currently chasing why websockets fails in the browser

**Comment by diasdavid** [comment]
more diagnoses 
- with webrtc works fine
- only fails browser websockets -> node.js websockets
- fails on multistream handshake (ms.handle), but only in the second one, when it tries to muxer upgrade, it works fine for the secio upgrade

**Comment by diasdavid** [comment]
I believe that secio might be mangling data between browser <-> node connections
node <-> node is fine
browser <-> browser is fine
browser <-> node without secio is fine

**Comment by diasdavid** [comment]
This is blocked by:

- https://github.com/libp2p/js-libp2p-secio/pull/30
- https://github.com/libp2p/js-libp2p-crypto/pull/29

**Comment by dignifiedquire** [comment]
Ready for review and unblocked with https://github.com/libp2p/js-libp2p-crypto/pull/29

File: test/websockets-only.js [hunk 2]
```diff
@@ -66,7 +66,7 @@ describe('libp2p-ipfs-browser (websockets only)', () => {
     })
   })
 
-  it('libp2p.dialByMultiaddr on Protocol nodeA to nodeB', (done) => {
+  it.skip('libp2p.dialByMultiaddr on Protocol nodeA to nodeB', (done) => {
```
Comment (diasdavid) [inline test/websockets-only.js]: ??

Comment (dignifiedquire) [inline test/websockets-only.js]: missed one -.-

Comment (dignifiedquire) [inline test/websockets-only.js]: this is not on my computer oO

**Pull request merged**